### PR TITLE
Skote2/working prototype

### PR DESCRIPTION
**Pull request by Skote2**
Currently I have a library linking to an executable and creating ascii images like so:
image

Clearly there are flaws and the API is not anywhere implemented like something we can integrate with ASCIIPlay, but this is a solid start and a working prototype.
This is leaking memory right now*** I am aware
It's in some pretty obvious "handed this off through a function and didn't free" kinda situations.

### CODE REVIEW

File: output.txt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+$WWWW&W###*oahkkkbbbbbbbbbbbbbbbbbbbbkkkhhhhhaaaaoooooaaaaaaaaaaaaaaaaaoo***##MMMMWWWWWMM##**oooaaahhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhhaaaoaaa
+$WWWW&WMM#*oahhkkbbbbbbbbbbbbbbbbbbbkkkkhhhhaaaaooooooooooaaaaaaaaoaoaooo**###MMMMWWWWWMM##**oooaahhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhhaaaaaa
+$WWWW&&WWM#ooahkkbbbbbbbbbbbbbbbbbbbkkkkhhhhaaaaoooooooooooooooooooooooo**###MMMWWWWWWWMM##**oooaahhhhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhaaaaaa
+$WWW&&&WMWM*oahkkbbbbbbbbbbbbbbbbbbkkkkhhhhhaaaaooooooooooooooooooooooo**####MMMWWWWWWWMM##**oooaahhhhhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhaaaaa
+$WWW&&&WMWW*ahhhkbbbbbbbddbbbbbbbbbkkkkkhhhhaaaaaaooooooooooooooooooooo**###MMMWWWWWWWWMM##**oooaahhhhhhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhaaaah
+$WMW&88&WW&#oaoahbbbbbbddddbbbbbbbbbkkkkhhhhaaaaaaaoooooooooooooo***oo***###MMMWWWWWWWWMM##**oooaaahhhhhhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkkkhhhhaaahh
+$WMW&88&WW&W*ooahkbbbbbddddbbbbbbbbbkkkkkhhhhaaaaaaoooooooooooooo***oo**####MMWWWWWWWWWMM##**oooaaahhhhhhhhhhhhhhhhhhhhhhhhhhhhkkkkkkkkkkkkkkkkkkkkkkkkhhhhaahhh
```
Comment (AngelOnFira) [inline output.txt:121]: Did you mean to commit this file?

File: test_images/shrek_army.png
Comment (AngelOnFira) [inline test_images/shrek_army.png:1]: This shrek file is thicc, maybe change to jpg if it needs to be added?

File: output.txt (hunk 1 above)
Comment (Skote2) [reply output.txt:121]: Oh that definitely shouldn't be here

File: test_images/shrek_army.png
Comment (Skote2) [reply test_images/shrek_army.png:1]: It doesn't , I can toss it

**Pull request merged**